src/pow/miner.cpp

```diff
@@ -203,23 +203,25 @@ void Miner::Run() {
                 b.SetMerkle();
             } else {
                 prevHash = selfChainHead_->GetHash();
+
+                auto tx = MEMPOOL->GetRedemptionTx(false);
+                if (tx && !tx->IsFirstRegistration()) {
+                    b.AddTransaction(std::move(tx));
+                }
+
                 if (distanceCal_.Full()) {
                     if (counter % 10 == 0) {
                         spdlog::info("Hashing power percentage {}",
                                      distanceCal_.Sum().GetDouble() / (distanceCal_.TimeSpan() + 1) /
                                          (double) (std::atomic_load(&chainHead_)->snapshot->hashRate + 1));
                     }
 
-                    auto tx = MEMPOOL->GetRedemptionTx(false);
-                    if (tx && !tx->IsFirstRegistration()) {
-                        b.AddTransaction(std::move(tx));
-                    }
-
                     auto allowed =
                         CalculateAllowedDist(distanceCal_, std::atomic_load(&chainHead_)->snapshot->hashRate);
                     b.AddTransactions(MEMPOOL->ExtractTransactions(prevHash, allowed, GetParams().blockCapacity));
-                    b.SetMerkle();
                 }
+
+                b.SetMerkle();
```

Comment:
I suggest combine `SetMerkle` line 203 and line 224 into one line before `Solve`